File: python/yl.py
import sys

import parser
import interpreter


if len(sys.argv) < 2:
    cli_scope = interpreter.Scope()
    while True:
        cmd = input('> ')
        ast = parser.parse(cmd)
        ret = interpreter.evaluate(ast, cli_scope, False)
        print(ret)
else:
    if sys.argv[1] == '-e':
        code = sys.argv[2]
    else:
        f = open(sys.argv[1], 'r')
        code = f.read()
    ast = parser.parse(code)
    ret = interpreter.evaluate(ast)
    exit(ret)

File: ruby/yl.rb
require_relative 'parser'
require_relative 'interpreter'

scope = Scope.new
if ARGV.empty?
  loop do
    print '> '
    cmd = gets
    ast = Parser.parse(cmd)
    # Parser.print_ast ast
    ret = AST.new(ast).evaluate(scope)
    puts ret
  end
else
  code = if ARGV[0] == '-e'
           ARGV[1]
         else
           File.read(ARGV[0])
         end
  ast = Parser.parse(code)
  ret = AST.new(ast).evaluate(scope)
  exit ret.to_i
end
